### DISCUSSION TITLE
Expansions on modified requests

### DISCUSSION THREAD

**Discussion by thisni1s** [pocketbase/pocketbase#2131] [Q&A] [answered]
Hello,

I've followed the answer in this [discussion post](https://github.com/pocketbase/pocketbase/discussions/1805) how to add some extra info to the list request answers.

For some reason modifying the answers in this way seems to stop the expansion of relation fields specified in the url parameters. Filtering still works though.

So how can i return the result with the expanded fields?

Any help would be much appreciated

**Comment by ganigeorgiev** [comment, marked as answer]
Could you provide a code sample showing what data you return in the response?

In general, if you want to enrich your custom records with "expand" relations, you can use the following helpers:

```go
// enable ?expand query param support for a single record
apis.EnrichRecord(c, app.Dao(), record)

// enable ?expand query param support for multiple records from the same collection
apis.EnrichRecords(c, app.Dao(), records)
```
_(you can find some information about the supported arguments and options in the [godoc of the methods](https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.13.4/apis#EnrichRecord))_

**Reply by thisni1s** [reply]
Im just experimenting with returning some static metadata in the response.

The collection being listed has a relation to another collection `tasks` which should be expanded in the response.

```
// main.go
package main

import (
	"log"
	"net/http"

	"github.com/labstack/echo/v5"
	"github.com/pocketbase/pocketbase"
	"github.com/pocketbase/pocketbase/apis"
	"github.com/pocketbase/pocketbase/core"
	"github.com/pocketbase/pocketbase/models"
	"github.com/pocketbase/pocketbase/tools/hook"
)

func main() {
	app := pocketbase.New()
	
	app.OnRecordsListRequest("work_entries").Add(func(e *core.RecordsListEvent) error {
		data := map[string]any{
                        "page":       e.Result.Page,
			"perPage":    e.Result.PerPage,
			"totalItems": e.Result.TotalItems,
			"totalPages": e.Result.TotalPages,
			"items":      e.Result.Items,
			"information":  map[string]any{
				"version": "v2",
				"picture-url": "",				 
			},
		}
		e.HttpContext.JSON(http.StatusOK, data)
		return hook.StopPropagation
	})
	
	if err := app.Start(); err != nil {
		log.Fatal(err)
	}
	
}
```

**Reply by ganigeorgiev** [reply]
You should be able to call `apis.EnrichRecords(e.HttpContext, app.Dao(), e.Result.Items.([]*models.Record))` before writing the response with the `e.HttpContext.JSON()` call.

**Reply by thisni1s** [reply]
That worked! Thank you so much!
